src/agent/communicator/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+cmake_minimum_required(VERSION 3.24)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_BUILD_TYPE RelWithDebInfo)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
+
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/../../vcpkg/scripts/buildsystems/vcpkg.cmake")
+set(VCPKG_MANIFEST_DIR ${CMAKE_SOURCE_DIR}/../../)
+
+project(Communicator)
+message(STATUS "******* Project Communicator *********")
+
+#########################################################################
+# Ensure that we do an out of source build
```

Comment:
What is this for?

Reply:
To ensure we make an out-of-source build and prevent from inadvertently running a cmake command from the wrong location. We'll remove it for now and bring it back down the road.